pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -190,11 +187,16 @@ public int removeIf(LongPairPredicate filter) {
      * @return a new list of keys with max provided numberOfItems (makes a copy)
      */
     public Set<LongPair> items(int numberOfItems) {
-        Set<LongPair> items = new HashSet<>();
+        return items(numberOfItems, (item1, item2) -> new LongPair(item1, item2));
+    }
+
+    @Override
+    public <T> Set<T> items(int numberOfItems, BiFunction<Long, Long, T> longPairConverter) {
+        Set<T> items = new HashSet<>();
```

Comment:
Is it necessary to return a set? typically the iteration methods are used to go through it inline without intermediary collection.

Reply:
yes, this method is used by [PersistentDispatcherMultipleConsumers](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java#L291) to prepare the sub-set of replay-messages and then it passes this set to `cursor.asyncReplayEntries` to read the entries. So, this method is not used for inline execution.